### PR TITLE
Ports atmos devices and tweaks from /tg/

### PR DESCRIPTION
**Pull request by BrumalLake**
## About The Pull Request
Ports [tgstation#53608](https://github.com/tgstation/tgstation/pull/53608), [tgstation#55926](https://github.com/tgstation/tgstation/pull/55926), [tgstation#53281](https://github.com/tgstation/tgstation/pull/53281), [tgstation#53362](https://github.com/tgstation/tgstation/pull/53362), [tgstation#53375](https://github.com/tgstation/tgstation/pull/53375), [tgstation#53632](https://github.com/tgstation/tgstation/pull/53632), [tgstation#53678](https://github.com/tgstation/tgstation/pull/53678), [tgstation#56611](https://github.com/tgstation/tgstation/pull/56611), and [tgstation#50150](https://github.com/tgstation/tgstation/pull/50150)
Temperature pump from tg
image
Transfers energy without mixing gas in one direction as long as the input is hotter than the output
[tgstation#59460](https://github.com/tgstation/tgstation/pull/59460) not ported cause we don't have a temperature cap as far as I know below overflow
Pressure valve
image
One way valve that releases pressure if the input has a higher pressure than its set value and its output gate
Temperature gate
image
Lets gas through if the temperature is higher/lower than set value depending on mode set through using a multitool on it
Only problem is that it gets stuck in closed systems because gates dont exchange gas when pressures are equal but temperatures arent
Probably fixed in another tg pr like passive vents
Passive gate more useful and has a max of 100 atmospheres now
~~Passive vents now keep exchanging even if internal and external pressures and equal, and equalize temperature and ratios better~~
- [x] temperature pump
- [x] pressure valve
- [x] temperature gate
- [x] [tgstation#53678](https://github.com/tgstation/tgstation/pull/53678)
- [x] ~~potentially [tgstation#50150](https://github.com/tgstation/tgstation/pull/50150) if i can figure out how to make it work with monstermos procs~~

## Why It's Good For The Game
More atmos devices, more versatile setups

## Changelog
:cl: lordScrubling, Ghilker, TetraK1, Dennok, Donkie
add: Temperature pump atmos device from tg
add: Pressure valve from atmos device tg
add: Temperature gate atmos device from tg
tweak: Passive gate pressure limit upped to 100 atmospheres
/:cl:

### CODE REVIEW

**Comment by BrumalLake** [comment]
gonna port the other devices and changes first

**Comment by Ghilker** [comment]
nice

**Review by utilbird — DISMISSED** [review]

**Comment by BrumalLake** [comment]
i dont remember requesting that
the fuck
whered the 20000 lines come from

**Comment by BrumalLake** [comment]
nothing atmos related is gonna get merged is it, until auxmos

**Comment by ike709** [comment]
> nothing atmos related is gonna get merged is it, until auxmos

Correct!

**Comment by BrumalLake** [comment]
well, the only thing i'd have to change with auxmos is revert the vents equalizing temperature, as the auxmos changes already seem to do it

**Comment by BrumalLake** [comment]
works fine from my testing
ready to work with auxmos

**Comment by BrumalLake** [comment]
?rebuild tgui

File: code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+		radio_connection = null
+	return ..()
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/update_icon_nopipes()
+	if(on && is_operational() && is_gas_flowing)
+		icon_state = "pvalve_flow"
+	else if(on && is_operational() && !is_gas_flowing)
+		icon_state = "pvalve_on"
+	else
+		icon_state = "pvalve_off"
```
Comment (utilbird) [inline code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm]:
```suggestion
	if(on && is_operational()) && is_gas_flowing)
		if(is_gas_flowing)
			icon_state = "pvalve_flow"
		else
			icon_state = "pvalve_on"
	else
		icon_state = "pvalve_off"
```

File: code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+	radio_connection.post_signal(src, signal, filter = RADIO_ATMOSIA)
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AtmosPump", name)
+		ui.open()
+
+/obj/machinery/atmospherics/components/binary/pressure_valve/ui_data()
+	var/data = list()
```
Comment (utilbird) [inline code/modules/atmospherics/machinery/components/binary_devices/pressure_valve.dm:95]:
```suggestion
	var/list/data = list()
```

File: code/modules/atmospherics/machinery/components/binary_devices/temperature_gate.dm [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+	if(inverted)
+		. += "The device settings can be restored if a multitool is used on it."
+	else
+		. += "The sensor's settings can be changed by using a multitool on the device."
+
+/obj/machinery/atmospherics/components/binary/temperature_gate/update_icon_nopipes()
+	if(on && is_operational() && is_gas_flowing)
+		icon_state = "tgate_flow-[set_overlay_offset(piping_layer)]"
+	else if(on && is_operational() && !is_gas_flowing)
+		icon_state = "tgate_on-[set_overlay_offset(piping_layer)]"
```
Comment (utilbird) [inline code/modules/atmospherics/machinery/components/binary_devices/temperature_gate.dm]:
```suggestion
	if(on && is_operational())
		if(is_gas_flowing)
			icon_state = "tgate_flow-[set_overlay_offset(piping_layer)]"
		else
			icon_state = "tgate_on-[set_overlay_offset(piping_layer)]"
```

File: code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+	update_parents()
+
+/obj/machinery/atmospherics/components/binary/temperature_pump/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
+		ui = new(user, src, "AtmosTempPump", name)
+		ui.open()
+
+/obj/machinery/atmospherics/components/binary/temperature_pump/ui_data()
+	var/data = list()
```
Comment (utilbird) [inline code/modules/atmospherics/machinery/components/binary_devices/temperature_pump.dm:70]:
```suggestion
	var/list/data = list()
```

**Review by utilbird — COMMENTED** [review]
auxmos is done so this should be good to go

**Comment by BrumalLake** [comment]
image

**Comment by ike709** [comment]
Needs to be tested with auxmos.

**Comment by BrumalLake** [comment]
soo testmerge or something?
the gas procs used should work exactly the same as before
apparently i tested it right after auxmos got merged as well after removing the redundant passive vents according to my comment above

**Review by Crossedfall — APPROVED** [review]
Tested and it works in staging. Needs wiki changes before merging.

**Comment by BrumalLake** [comment]
https://wiki.beestation13.com/view/Guide_to_atmospherics
wiki updated just no images right now

**Comment by MCterra10** [comment]
isn't temp pump just the heat exchanger but on one tile instead of two?

**Pull request merged**